core/src/main/java/org/infinispan/metadata/EmbeddedMetadata.java

```diff
@@ -205,6 +210,11 @@ public Metadata.Builder builder() {
          return super.builder().lifespan(lifespan).maxIdle(maxIdle);
       }
 
+      @Override
+      public boolean isEmpty() {
+         return super.isEmpty() && lifespan < 0 && maxIdle < 0;
```

Comment:
shouldn't be `<= 0`?

Reply:
No, 0 has a different meaning or expiration so it has to be <

Reply:
what does `0` means? Expire immediately?